src/mscorlib/shared/System/Globalization/Extensions.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace System.Globalization
+{
+    public static class GlobalizationExtensions
+    {
+        public static StringComparer GetStringComparer(this CompareInfo compareInfo, CompareOptions options)
+        {
+            if (compareInfo == null)
+            {
+                throw new ArgumentNullException(nameof(compareInfo));
+            }
+
+            if (options == CompareOptions.Ordinal)
+            {
+                return StringComparer.Ordinal;
+            }
+
+            if (options == CompareOptions.OrdinalIgnoreCase)
+            {
+                return StringComparer.OrdinalIgnoreCase;
+            }
+
+            if ((options & CultureAwareComparer.ValidCompareMaskOffFlags) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
+            }
+
+            return new CultureAwareComparer(compareInfo, options);
+        }
+    }
+
+    internal sealed class CultureAwareComparer : StringComparer
```

Comment:
This should be deleted because of we want to use the CultureAwareComparer that is in CoreLib already.

Reply:
But that takes CultureInfo as input

Reply:
@Anipik you need to delete CultureAwareComparer from here and introduce a new constructor which takes CompareInfo in [https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/StringComparer.cs#L175](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/StringComparer.cs#L175) 

---
In reply to: [169498383](https://github.com/dotnet/coreclr/pull/16459#discussion_r169498383) [](ancestors = 169498383)